src/main/java/com/minecolonies/coremod/entity/ai/citizen/miner/EntityAIStructureMiner.java

```diff
@@ -1306,21 +1321,6 @@ private Node createNewNodeInDirectionFromNode(@NotNull final Node start, final i
         return node;
     }
 
-    @NotNull
-    private static Node.NodeType getRandomNodeType()
-    {
-        final int roll = new Random().nextInt(100);
-        if (roll > 50)
-        {
-            return Node.NodeType.TUNNEL;
-        }
-        if (roll > 20)
-        {
-            return Node.NodeType.BEND;
-        }
-        return Node.NodeType.CROSSROAD;
-    }
-
     private Node findNodeOnLevel(@NotNull final Level currentLevel)
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The Cyclomatic Complexity of this method "findNodeOnLevel" is 16 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)